src/resolvers/platform/WindowsSecurity.js

```diff
@@ -62,7 +62,25 @@ export default {
 
   async applications (root, args, context) {
     const device = await kmd('os', context)
-    const apps = await Device.applications(root, args, context)
+    const DEFAULT_REG_PATH = 'HKLM\\\\Software\\\\Microsoft\\\\Windows\\\\CurrentVersion\\\\Uninstall'
+
+    // gather set of optional registry path overrides from policy
+    const overrides = new Set()
+    args.applications.map(({ paths = {} }) => {
+        overrides.add(paths.win32 || DEFAULT_REG_PATH)
+    })
+
+    const registry_paths = Array.from(overrides)
```

Comment:
Instead of searching for all apps at the unique paths provided for applications, this should be simplified to search for the given application at the provided (or default) path. One of the reasons for being able to specify a paths for an application is to ensure it is installed at the preferred location. If we grab all apps at all locations we won't know if this requirement was satisfied. This will also match the logic used on mac

Reply:
@nathancharles the intended functionality here probably needs to be discussed a bit further. The GQL `applications` policy supports a `paths` argument, but it is unclear what this should mean.

On Mac, it makes sense for `paths` to refer to the actual filesystem path to a `.app` folder, which is what the `sources/darwin/app.sh` kmd expects. It searches for a singular application at an (overridable) path. It makes sense that Mac users would want to verify that the application was installed at a particular filesystem path, perhaps in keeping with their own system configuration guidelines, etc.

On Windows, applications installs are tracked in the central Windows registry, and the location of installed components in the filesystem may be scattered quite a bit amongst several folders and/or hidden system locations. 9 times out of 10, it is likely that Stethoscope users will want to use the default *registry* path needed for `sources/win32/apps.sh` to look for installed applications. If a `paths.win32` argument is specified at all, it should probably mean "override the default win32 registry location when determining if this particular application is installed". It makes less sense, to me, on win32 to ensure that applications assets are installed at particular filesystem locations. 

So, we have a pretty big logic difference in how applications are discovered on Mac vs Windows. The Mac resolver is a bit like "Go Fish", where it calls `defaults read` a bunch of times, and the Windows resolver gets a (potentially large) list via `reg query` and checks to see if each application specified in the policy is on the list. The `reg query` operation is expensive, and it doesn't make sense to run it any more than we have to.

The reasoning above led to my (current) approach of generating an installed applications list with a minimal number of calls to `reg query` (should be exactly one, unless Stethoscope users need to check for a non-standard registry location).

I think my understanding here needs to be clarified before I can proceed. Curious what your thoughts are.

Reply:
Ah,  this is my misunderstanding of Windows 😊
This makes a lot more sense now

Reply:
@nathancharles, thanks. Are you on board with overloading the policy `application.paths` variable to mean filesystem paths in the context of Darwin (and possibly Linux), and *registry* paths in the context of Win32?

If so, is there anything else you'd like to see here in this PR?

Reply:
Yeah, that makes sense for me. Please do update the docs, especially so we can call out the behavior difference for windows.